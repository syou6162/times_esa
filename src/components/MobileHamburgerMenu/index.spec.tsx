// @vitest-environment jsdom
import { fireEvent, render } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { MobileHamburgerMenu, MobileHamburgerMenuProps } from '.';

describe('MobileHamburgerMenuが正しく動作する', () => {
  it('閉じた状態で表示される', () => {
    const mockOnToggle = vi.fn();
    const props: MobileHamburgerMenuProps = {
      isOpen: false,
      onToggle: mockOnToggle,
      children: <div>サイドバーコンテンツ</div>,
    };

    const renderResult = render(<MobileHamburgerMenu {...props} />);

    expect(renderResult.asFragment()).toMatchSnapshot();
    expect(renderResult.getByLabelText('メニューを開く')).toBeTruthy();
    expect(renderResult.queryByText('サイドバーコンテンツ')).toBeNull();
  });

  it('開いた状態で表示される', () => {
    const mockOnToggle = vi.fn();
    const props: MobileHamburgerMenuProps = {
      isOpen: true,
      onToggle: mockOnToggle,
      children: <div>サイドバーコンテンツ</div>,
    };

    const renderResult = render(<MobileHamburgerMenu {...props} />);

    expect(renderResult.asFragment()).toMatchSnapshot();
    expect(renderResult.getByText('サイドバーコンテンツ')).toBeTruthy();
  });

  it('ハンバーガーボタンをクリックするとonToggleが呼ばれる', () => {
    const mockOnToggle = vi.fn();
    const props: MobileHamburgerMenuProps = {
      isOpen: false,
      onToggle: mockOnToggle,
      children: <div>サイドバーコンテンツ</div>,
    };

    const renderResult = render(<MobileHamburgerMenu {...props} />);

    fireEvent.click(renderResult.getByLabelText('メニューを開く'));
    expect(mockOnToggle).toHaveBeenCalledOnce();
  });

  it('複数の子要素を正しく表示する', () => {
    const mockOnToggle = vi.fn();
    const props: MobileHamburgerMenuProps = {
      isOpen: true,
      onToggle: mockOnToggle,
      children: (
        <>
          <div>タイトル</div>
          <div>リスト項目1</div>
          <div>リスト項目2</div>
        </>
      ),
    };

    const renderResult = render(<MobileHamburgerMenu {...props} />);

    expect(renderResult.getByText('タイトル')).toBeTruthy();
    expect(renderResult.getByText('リスト項目1')).toBeTruthy();
    expect(renderResult.getByText('リスト項目2')).toBeTruthy();
  });
});