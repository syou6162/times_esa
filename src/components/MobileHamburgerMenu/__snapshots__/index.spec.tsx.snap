// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MobileHamburgerMenuが正しく動作する > 閉じた状態で表示される 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1406zqk"
  >
    <button
      aria-label="メニューを開く"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1s19k05-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`MobileHamburgerMenuが正しく動作する > 開いた状態で表示される 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1406zqk"
  >
    <button
      aria-label="メニューを開く"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1s19k05-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
